lib/authorizer/oauth2.js

```diff
@@ -58,6 +59,34 @@ module.exports = {
      * @param {AuthHandlerInterface~authPreHookCallback} done -
      */
     pre: function (auth, done) {
+        const authId = auth.get('authId'),
+            refreshTokenHelper = auth.get('refreshTokenHelper');
+
+        // Go into the refresh token flow ONLY IF we have all necessary information present
+        if (authId && refreshTokenHelper && _.isFunction(refreshTokenHelper.refreshToken)) {
```

Comment:
Why not just bail out if `authId` or `refreshTokenHelper` is not set?
```suggestion

        if (!(authId && refreshTokenHelper && _.isFunction(refreshTokenHelper.refreshToken))) {
        	return done(null, Boolean(auth.get('accessToken')));
        }
```